src/render/draw_symbol.js

```diff
@@ -233,18 +301,19 @@ function drawLayerSymbols(painter, sourceCache, layer, coords, isText, translate
         const labelPlaneMatrix = symbolProjection.getLabelPlaneMatrix(coord.posMatrix, pitchWithMap, rotateWithMap, painter.transform, s);
         const glCoordMatrix = symbolProjection.getGlCoordMatrix(coord.posMatrix, pitchWithMap, rotateWithMap, painter.transform, s);
 
+        const hasVariableAnchors = variablePlacement && bucket.hasTextData();
+        const updateTextFitIcon = layer.layout.get('icon-text-fit') !== 'none' &&
+            hasVariableAnchors &&
+            bucket.hasIconData();
+
         if (alongLine) {
             symbolProjection.updateLineLabels(bucket, coord.posMatrix, painter, isText, labelPlaneMatrix, glCoordMatrix, pitchWithMap, keepUpright);
-        } else if (isText && size && variablePlacement) {
-            const tileScale = Math.pow(2, tr.zoom - tile.tileID.overscaledZ);
-            updateVariableAnchors(bucket, rotateWithMap, pitchWithMap, variableOffsets, symbolSize,
-                                  tr, labelPlaneMatrix, coord.posMatrix, tileScale, size);
-        } else if (isText && size && bucket.allowVerticalPlacement) {
+        } else if (isText && size && bucket.allowVerticalPlacement && !variablePlacement) {
```

Comment:
Perhaps I'm missing something but why the `&& !variablePlacement` here? IIRC we can have both vertical & variable placement enabled at the same time, are we accounting for that case?

Reply:
No, we're not. In the interest of time time this PR only handles `icon-text-fit` rendering for horizontal variable anchor labels.

Reply:
:ok_hand: Maybe we should ticket that out